tests/units/test_state.py

```diff
@@ -2513,10 +2513,7 @@ def test_mutable_copy(mutable_state: MutableTestState, copy_func: Callable):
         assert getattr(ms_copy, attr) is not getattr(mutable_state, attr)
     ms_copy.custom.array.append(42)
     assert "custom" in ms_copy.dirty_vars
-    if copy_func is copy.copy:
```

Comment:
i can implement `__copy__` for `BaseState` if you want to keep this behavior

Reply:
i dont think anything is depending on this behavior, so we're fine to simplify the test logic -- famous last words 🫣